### PR TITLE
Use libblockdev FS plugin for filesystem operations

### PR DESCRIPTION
**Pull request by vojtechtrefny**
With libblockdev 3.0 now released and required in Blivet, we can now start using the FS plugin for various filesystem operations we previously used the command line tools directly.

Most of the changes are just replacing the function calls with libblockdev. One major change is removing some filesystem support. We'll remove support for creating and managing of ReiserFS, JFS and HFS. These filesystems are either unsupported (ReiserFS userspace tools were removed from Fedora and ReiserFS is also deprecated in the upstream kernel) or very old and not used (HFS is supported only on PPC Mac which is no longer supported in Fedora, we are not removing HFS+ support).

TODO:
- [x] FS min size support -- blivet uses different tools for getting filesystem min size (`ntfsresize` used by blivet vs `ntfsinfo` used by libblockdev to get filesystem free space), we need to add support to libblockdev first
- [x] swap UUID and label format check, libblockdev support pending release: https://github.com/storaged-project/libblockdev/pull/978
- [ ]  fsck support -- blivet uses error codes for custom error messages, which libblockdev doesn't support

### CODE REVIEW

File: blivet/tasks/fsmkfs.py [hunk 1]
```diff
@@ -293,67 +237,112 @@ def args(self):
...
+                                   " creation of filesystem %s. UUID format"
+                                   " is unacceptable for this filesystem."
+                                   % (self.fs.uuid, self.fs.type))
+        if label and self.fs.label and not self.fs.label_format_ok(self.fs.label):
+            raise FSWriteLabelError("Choosing not to apply label (%s) during"
+                                    " creation of filesystem %s. Label format"
+                                    " is unacceptable for this filesystem."
+                                    % (self.fs.label, self.fs.type))
+
+        # FIXME: this will be fixed in libblockdev
```
Comment (vojtechtrefny) [inline blivet/tasks/fsmkfs.py:275]: libblockdev PR: https://github.com/storaged-project/libblockdev/pull/981

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Comment by vojtechtrefny** [comment]
Marking as ready for review, we are still missing some features, but these can be implemented later in a separated PR without blocking this.

File: blivet/tasks/fswriteuuid.py [hunk 2]
```diff
@@ -2,67 +2,50 @@
...
 from . import availability
 from . import fstask
 from . import task
 
+import gi
+gi.require_version("BlockDev", "3.0")
+from gi.repository import BlockDev
+
 
 @add_metaclass(abc.ABCMeta)
```
Comment (japokorn) [inline blivet/tasks/fswriteuuid.py:16]: Is the decorator still needed here?

File: blivet/tasks/fsresize.py [hunk 3]
```diff
@@ -38,13 +34,15 @@
 import logging
 log = logging.getLogger("blivet")
 
+import gi
+gi.require_version("BlockDev", "3.0")
+from gi.repository import BlockDev
+
 
 @add_metaclass(abc.ABCMeta)
```
Comment (japokorn) [inline blivet/tasks/fsresize.py:42]: Is the decorator still needed here? More of the edited functions have the abc decorator left in them even though abc is no longer used in them.

**Review by japokorn — APPROVED** [review]
Looks good to me.

**Pull request merged**